include/IECoreGL/RawTexture.h

```diff
@@ -0,0 +1,86 @@
+//  met:
+//
+//     * Redistributions of source code must retain the above copyright
+//       notice, this list of conditions and the following disclaimer.
+//
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//
+//     * Neither the name of Image Engine Design nor the names of any
+//       other contributors to this software may be used to endorse or
+//       promote products derived from this software without specific prior
+//       written permission.
+//
+//  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS
+//  IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+//  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+//  PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR
+//  CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+//  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+//  PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+//  PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF
+//  LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING
+//  NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+//  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef IECOREGL_RAWTEXTURE_H
+#define IECOREGL_RAWTEXTURE_H
+
+#include "IECoreGL/Export.h"
+#include "IECoreGL/Texture.h"
+
+namespace IECoreGL
+{
+
+IE_CORE_FORWARDDECLARE( Shader );
+
+/// The RawTexture class represents a texture that has
+/// been allocated manually using glGenTextures.
+/// This class will take care of binding the texture,
+/// and of deleting it with glDeleteTextures when this
+/// class is destructed.
+///
+/// IMPORTANT \todo - Currently not supported by Texture::ScopedBinding.
+/// Only supported by Texture::bind() and Shader::Setup.
+///
+/// Once we can break ABI, we need to add targetType to Texture, and
+/// then this will become a passthrough shim to Texture which is
+/// correctly supported by Texture::ScopedBinding
+class IECOREGL_API RawTexture : public Texture
+{
+	public :
+
+		IE_CORE_DECLARERUNTIMETYPEDEXTENSION( IECoreGL::RawTexture, RawTextureTypeId, Texture );
+
+		/// Creates a texture object to wrap the already created
+		/// GL texture specified, with a targetType that is expected to be
+		/// one of GL_TEXTURE_1D, GL_TEXTURE_2D or GL_TEXTURE_3D
+		RawTexture( GLuint texture, GLenum targetType )
+			: Texture( texture ), m_targetType( targetType )
+		{
+		};
+
+	private :
+
+		// Needed to do some hackery in Texture and Shader since the base class doesn't yet know about
+		// targetType
+		friend Texture;
+		friend Shader;
```

Comment:
I'm really confused by this comment and the next one ... since you decided we wouldn't go with the version of the code that uses RawTexture.  I was freaked out that I may have uploaded the wrong version, but looking at this PR now, I only see one commit, and it doesn't add RawTexture ... I guess you were looking at something out of date?

Reply:
Oh, yeah, that is freaky. Those are comments I made during my first review - I definitely didn't intentionally repeat them for this one, as there was only that one commit to look at. I guess we put it down to a GitHub bug...

Just to be completely clear, this LGTM, so feel free to merge once you've retargeted it to `RB-10.4`.